lib/collection/src/update_handler.rs

```diff
@@ -376,6 +380,48 @@ impl UpdateHandler {
         handles
     }
 
+    /// Ensure there is at least one appendable segment with enough capacity
+    ///
+    /// If there is no appendable segment, or all are at or over capacity, a new empty one is
+    /// created.
+    ///
+    /// Capacity is determined based on `optimizers.max_segment_size_kb`.
+    pub(super) fn ensure_appendable_segment_with_capacity(
+        segments: &LockedSegmentHolder,
+        segments_path: &Path,
+        collection_params: &CollectionParams,
+        thresholds_config: &OptimizerThresholds,
+    ) -> OperationResult<()> {
+        let no_segment_with_capacity = {
+            let segments_read = segments.read();
+            segments_read
+                .appendable_segments_ids()
+                .into_iter()
+                .filter_map(|segment_id| segments_read.get(segment_id))
+                .all(|segment| {
+                    let max_vector_size_bytes = segment
+                        .get()
+                        .read()
+                        .max_available_vectors_size_in_bytes()
+                        .unwrap_or_default();
+                    let max_segment_size_bytes = thresholds_config
+                        .max_segment_size_kb
+                        .saturating_add(segment::common::BYTES_IN_KB);
```

Comment:
Is this right? Should it be `saturating_mul`? 🤔

Reply:
Would be kinda fun if this fix perf degradation you've seen in your tests... 😂

Reply:
Nice catch!

Reply:
:clown_face: 

Thanks!

Reply:
It can explain poor benchmark speed, if segment max size is low, we would create too many segments